pkg/api/cachedcsi.go

```diff
@@ -0,0 +1,203 @@
+package api
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"os"
+	"path"
+	"path/filepath"
+	"syscall"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"github.com/gadget-inc/dateilager/internal/key"
+	"github.com/gadget-inc/dateilager/internal/logger"
+	"github.com/gadget-inc/dateilager/pkg/version"
+	"github.com/golang/protobuf/ptypes/wrappers"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	DriverName = "com.gadget.dateilager.cached"
+)
+
+// GetPluginInfo returns metadata of the plugin
+func (c *Cached) GetPluginInfo(ctx context.Context, req *csi.GetPluginInfoRequest) (*csi.GetPluginInfoResponse, error) {
+	resp := &csi.GetPluginInfoResponse{
+		Name:          DriverName,
+		VendorVersion: version.Version,
+	}
+
+	return resp, nil
+}
+
+// GetPluginCapabilities returns available capabilities of the plugin
+func (c *Cached) GetPluginCapabilities(ctx context.Context, req *csi.GetPluginCapabilitiesRequest) (*csi.GetPluginCapabilitiesResponse, error) {
+	resp := &csi.GetPluginCapabilitiesResponse{
+		Capabilities: []*csi.PluginCapability{},
+	}
+
+	return resp, nil
+}
+
+// Probe returns the health and readiness of the plugin
+func (c *Cached) Probe(ctx context.Context, req *csi.ProbeRequest) (*csi.ProbeResponse, error) {
+	return &csi.ProbeResponse{
+		Ready: &wrappers.BoolValue{
+			Value: true,
+		},
+	}, nil
+}
+
+// NodeGetCapabilities returns the supported capabilities of the node server
+// this driver has no capabilities like expansion or staging, because we only use it for node local volumes
+func (c *Cached) NodeGetCapabilities(ctx context.Context, req *csi.NodeGetCapabilitiesRequest) (*csi.NodeGetCapabilitiesResponse, error) {
+	nscaps := []*csi.NodeServiceCapability{}
+
+	return &csi.NodeGetCapabilitiesResponse{
+		Capabilities: nscaps,
+	}, nil
+}
+
+// NodeGetInfo returns the supported capabilities of the node server. This
+// Usually, a CSI driver would return some interesting stuff about the node here for the controller to use to place volumes, but because we're only supporting node local volumes, we return something very basic
+func (c *Cached) NodeGetInfo(ctx context.Context, req *csi.NodeGetInfoRequest) (*csi.NodeGetInfoResponse, error) {
+	return &csi.NodeGetInfoResponse{
+		NodeId:            first(os.Getenv("NODE_NAME"), "dev"),
```

Comment:
This env var was not set on a random pod I checked.

Reply:
You can probably enforce it by turning a K8S attribute into an env var: https://github.com/gadget-inc/dateilager/pull/88/files#diff-0f2bf9cc69968218e9a4568b38292597ab5690e77603cfd36db60661e703f22cR30-R34

Reply:
Yep we gotta do exactly that.